docs/development/making_pr.rst

```diff
@@ -0,0 +1,166 @@
+
+Switch to the :code:`develop` branch and merge the changes from
+:code:`change-one` branch to the :code:`develop` branch. Delete the
+:code:`change-one` branch, if you do not plan to use it anymore.
+
+.. code:: sh
+
+  git checkout develop # switching to develop branch
+  git merge change-one # merging the changes from change-one branch
+  git branch -d change-one # deleting the change-one branch
+
+:code:`git merge` incorporates changes from the named commits (since the time their histories
+diverged from the current branch) into the current branch. This command is
+used by `git pull` to incorporate changes from another repository and can be
+used by hand to merge changes from one branch into another.
+
+Assume the following history exists and the current branch you are on
+is :code:`develop`:
+
+.. code:: sh
+
+        A---B---C change-one
+       /
+  D---E---F---G develop
+
+Then :code:`git merge change-one` will replay the changes made on the
+:code:`change-one` branch since it diverged from :code:`develop` (i.e., E)
+until its current commit (C) on top of
+:code:`develop`, and record the result in a new commit along with the names of
+the two parent commits and a log message from the user describing the changes.
+
+.. code:: sh
+
+	  A---B---C change-one
+	 /         \
+    D---E---F---G---H develop
+
+.. note::
+  It is not necessary to merge :code:`change-one` branch into
+  :code:`develop` branch in order to open a pull request. It is perfectly fine
+  to open a pull request from the :code:`change-one` branch.
+
+Rebasing
+~~~~~~~~~
+
+Rebasing is the process of moving or combining a sequence of commits to a new
+base commit. Rebasing is most useful and easily visualized in the context of a
+feature branching workflow.
+
+Assume the following history exists:
+
+.. code:: sh
+
+          A---B---C change-one
+         /
+    D---E---F---G develop
+
+From this point, the result of either of the following commands:
+
+.. code:: sh
+
+  git rebase develop
+  git rebase develop change-one
+
+would be:
+
+.. code:: sh
+
+                    A`--B`--C` change-one
+                 /
+    D---E---F---G develop
+
+.. note::
+  :code:`A` and :code:`A`` represents the same set of changes, but have
+  different committer information. 
+
+.. warning::
+  The golden rule of git rebase is to never use it on public branches.
+
+Pushing the changes to GitHub Fork
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First get any updates made in the upstream repository.
+
+.. code:: sh
+
+  git pull upstream develop
+
+Once your changes are committed, push the changes to your GitHub fork.
+
+.. code:: sh
+
+  git push origin <branch-name>
+
+Making a Pull Request to get Your Changes Merged in :code:`develop` branch
+-------------------------------------------------------------------------------
+
+1. Through GitHub make a pull request from the branch that you commited your
+code to.
+
+2. Once PR is made, the s/Travis CI/Circle CI build server checks all tests
```

Comment:
Ah sorry - for future reference `s/old text/new text/` means "replace `old text` with `new text`". So here I was saying replace "Travis CI" with "Circle CI" (since we only use Circle CI).